app/controllers/saml_idp_controller.rb

```diff
@@ -148,10 +148,24 @@ def saml_response
     encode_response(
       current_user,
       authn_context_classref: requested_authn_context,
-      reference_id: auth_identity.session_uuid.gsub(/^_/, '')
+      reference_id: auth_identity.session_uuid.gsub(/^_/, ''),
+      encryption: saml_response_encryption_opts
     )
   end
 
+  def saml_response_encryption_opts
+    encryption_opts = nil
+    if current_user.last_identity
+      sp = ServiceProvider.new(current_user.last_identity.service_provider)
+      encryption_opts = {
+        cert: OpenSSL::X509::Certificate.new(sp.cert),
+        block_encryption: sp.block_encryption,
```

Comment:
Why happens with for the service provider without block_encryption set?  (looks like its only set for a couple of them).  
Is the intent to allow service providers to choose the algo, or opt of of encryption completely?